packages/ui-react-utils/src/safeCloneElement.ts

```diff
@@ -36,11 +36,9 @@ import { createChainedFunction } from '@instructure/ui-utils'
  * @param props Props of the element
  * @param children
  */
-function safeCloneElement<
-  P extends Record<string, any> & { style?: any } & ClassAttributes<P>
->(
+function safeCloneElement<P extends Record<string, any> & { style?: any }>(
   element: ReactElement<P> & { ref?: any },
-  props: P,
+  props: P & { ref?: any } & ClassAttributes<any>,
```

Comment:
Instead of `ClassAttributes` you can use here `Attributes`, since you define `ref` as `any`:
`props: P & { ref?: any } & Attributes`

Reply:
@matyasf  I ended up refactoring it again, and I think this time I got it finally :D I managed to infer the props and ref types from the element, and can finally return the original element type too.... Can you please re-check it?